app/controllers/application_controller.rb

```diff
@@ -5,6 +5,137 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions 
+    set :session_secret, 'super secret'
   end
 
+  get '/' do 
+  	erb :index
+	end
+
+	get '/signup' do
+		if !session[:id].nil?
+			redirect '/tweets'
+		end
+		erb :"users/create_user"
+	end
+
+	post '/signup' do
+		if params[:username].empty?||params[:password].empty?||params[:email].empty?
+			redirect '/signup'
+		else
+			@user = User.create(params)
+			session[:id]= @user.id
+			redirect '/tweets'
+		end
+	end
+
+	get '/login' do
+		if !session[:id].nil?
+			redirect '/tweets'
+		end
+		erb :'/users/login'
+	end
+
+	post '/login' do
+		@user =User.find_by(:username=> params[:username])
+		
+		if @user && @user.authenticate(params[:password])
+			session[:id] = @user.id
+			redirect '/tweets'
+		else
+			redirect '/login'
+		end
+	end
+
+	get '/tweets' do
```

Comment:
This controller is overloaded. Consider separating your concerns by splitting some of these actions into separate controllers (i.e. a users controller and a tweets controller).